### PR TITLE
feat: add ability to specify branches for (any)-file-content rules

### PR DESCRIPTION
**Pull request by Brend-Smits**
Signed-off-by: Brend Smits <brend.smits@philips.com>

## Motivation

Some rules require you to check not just the default branch, but also other branches. 

## Proposed Changes

This is now possible by using the 'branches' option. Currently only works for the any-file-contents and file-contents rule. 
The "default" branch name is reserved for the default branch that is retrieved when cloning the repository (in most cases, this would be main, master, or develop).

Small to-do's: 

- [x] Tidy up the code and possibly split it into a new function to later on re-use for other rules.

Note:
Some status-checks fail, we should update dependencies and merge upstream into the fork so we stay up to date. This is a task for later though.
Tested locally and internally on Continuous Compliance and everything works.

### CODE REVIEW

**Comment by Brend-Smits** [comment]
This is ready for review.

File: rules/file-contents-config.json [hunk 1]
```diff
@@ -11,6 +11,11 @@
       "type": "array",
       "items": { "type": "string" }
     },
+    "branches": {
+      "type": "array",
+      "items": { "type": "string" },
+      "default": ["default"]
```
Comment (JeroenKnoops) [inline rules/file-contents-config.json:17]: `default` is in this case not the actual branch-name but a magic label, the default of the repository.

Did I understand that correctly?

File: rulesets/debug.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "rules": {
+    "philips-internal-com-migrationv2": {
+      "level": "warning",
+      "rule": {
+        "type": "file-not-contents",
+        "options": {
+          "globsAll": ["CNAME"],
+          "content": ".*\\.(philips-internal.com)",
+          "fail-on-non-existent": false,
+          "branches": ["default", "origin/gh-pages"]
```
Comment (JeroenKnoops) [inline rulesets/debug.json:18]: In this case, both the default branch (f.e. `main`) and `origin/gh-pages` are checked!

**Review by JeroenKnoops — APPROVED** [review]

File: rules/file-contents-config.json (hunk 1 above)
Comment (Brend-Smits) [reply rules/file-contents-config.json:17]: Indeed!

File: rulesets/debug.json (hunk 2 above)
Comment (Brend-Smits) [reply rulesets/debug.json:18]: Yes :) That is correct.

**Pull request merged**